src/vs/workbench/contrib/terminal/browser/terminalIconPicker.ts

```diff
@@ -51,22 +51,34 @@ export class TerminalIconPicker extends Disposable {
 		});
 	}
 
-	async pickIcons(): Promise<ThemeIcon | undefined> {
+	async pickIcons(source: 'commandPalette' | 'terminalTab'): Promise<ThemeIcon | undefined> {
 		const dimension = new Dimension(486, 260);
 		return new Promise<ThemeIcon | undefined>(resolve => {
 			this._register(this._iconSelectBox.onDidSelect(e => {
 				resolve(e);
 				this._iconSelectBox.dispose();
 			}));
 			this._iconSelectBox.clearInput();
+
 			const body = getActiveDocument().body;
 			const bodyRect = body.getBoundingClientRect();
+
+			let targetX = bodyRect.left + (bodyRect.width - dimension.width) / 2;
+			let targetY = bodyRect.top + this._layoutService.activeContainerOffset.quickPickTop - 2;
+
+			if (source === 'terminalTab') {
+				// Example positioning: near top-left where terminal tabs usually are
+				targetX = bodyRect.left + 50; // adjust as needed
+				targetY = bodyRect.top + 100;  // adjust as needed
+			}
+			// else keep default centered position for commandPalette
```

Comment:
This doesn't solve the problem at all and just positions it randomly based on arbitrary values. Getting the real values is literally everything for this issue. (50, 100) puts it randomly in the top left corner 🤔